### PR TITLE
refactor(core)!: Make `getBinaryStream` async

### PR DESCRIPTION
**Pull request by ivov**
Story: [PAY-846](https://linear.app/n8n/issue/PAY-846) | Related: https://github.com/n8n-io/n8n/pull/7225

For the S3 backend for external storage of binary data and execution data, the `getAsStream` method in the binary data manager interface used by FS and S3 will need to become async. This is a breaking change for nodes-base.

### CODE REVIEW

File: packages/nodes-base/nodes/HttpRequest/GenericFunctions.ts [hunk 1]
```diff
@@ -138,21 +138,31 @@ export const binaryContentTypes = [
...
-export const prepareRequestBody = (
+export async function reduceAsync<T, R>(
+	arr: T[],
+	reducer: (acc: Awaited<Promise<R>>, cur: T) => Promise<R>,
+	init: Promise<R> = Promise.resolve({} as R),
+): Promise<R> {
+	return arr.reduce(async (promiseAcc, item) => {
+		return reducer(await promiseAcc, item);
+	}, init);
+}
```
Comment (ivov) [inline packages/nodes-base/nodes/HttpRequest/GenericFunctions.ts:151]: Decided to add a adapter in the interest of time and to minimize the refactor, but generally speaking I find the heavy use of reducers in the HTTP node rather hard to follow.

File: packages/nodes-base/nodes/HttpRequest/test/utils/utils.test.ts [hunk 2]
```diff
@@ -11,15 +11,13 @@ describe('HTTP Node Utils, prepareRequestBody', () => {
 		];
 		const defaultReducer: BodyParametersReducer = jest.fn();
 
-		prepareRequestBody(bodyParameters, 'json', 3, defaultReducer);
+		await prepareRequestBody(bodyParameters, 'json', 3, defaultReducer);
 
 		expect(defaultReducer).toBeCalledTimes(1);
-		expect(defaultReducer).toBeCalledWith({}, { name: 'foo.bar', value: 'baz' }, 0, [
-			{ name: 'foo.bar', value: 'baz' },
-		]);
```
Comment (ivov) [inline packages/nodes-base/nodes/HttpRequest/test/utils/utils.test.ts:19]: It would seem the second pair of args (injected by Jest) are no longer compatible with the async reducer. 

From what I can tell, Jest tracks how many times the function `defaultReducer` has been called, and provides the call index as an argument. And `[{ name: 'foo.bar', value: 'baz' }]` is the list of arguments that `defaultReducer` was called with.

Since these values are already being checked in the first pair of args, to keep this moving I decided to remove the second pair. Let me know if they are needed or you have any insights on why they're incompatible with async.

File: packages/cli/src/WebhookHelpers.ts [hunk 3]
```diff
@@ -506,15 +506,15 @@ export async function executeWebhook(
...
 					}
 
 					const binaryData = (response.body as IDataObject)?.binaryData as IBinaryData;
 					if (binaryData?.id) {
 						res.header(response.headers);
-						const stream = Container.get(BinaryDataService).getAsStream(binaryData.id);
+						const stream = await Container.get(BinaryDataService).getAsStream(binaryData.id);
 						void pipeline(stream, res).then(() =>
 							responseCallback(null, { noWebhookResponse: true }),
 						);
```
Comment (netroy) [inline packages/cli/src/WebhookHelpers.ts:520]: since the outer function is `async` now, can you please also update this to 

```suggestion
						await pipeline(stream, res);
						responseCallback(null, { noWebhookResponse: true });
```

File: packages/nodes-base/nodes/HttpRequest/GenericFunctions.ts (hunk 1 above)
Comment (netroy) [inline packages/nodes-base/nodes/HttpRequest/GenericFunctions.ts:151]: This is definitely not easy to follow :/

**Review by netroy — CHANGES_REQUESTED** [review]
besides the first comment, LGTM.

File: packages/nodes-base/nodes/HttpRequest/GenericFunctions.ts (hunk 1 above)
Comment (ivov) [reply packages/nodes-base/nodes/HttpRequest/GenericFunctions.ts:151]: Agree, this is to minimize changes - to make all this easier to follow requires a bigger refactor.

**Review by netroy — APPROVED** [review]

**Comment by janober** [comment]
Got released with [`n8n@1.9.0`](https://github.com/n8n-io/n8n/releases/tag/n8n%401.9.0)
